src/Tuura/Concept/FSM/Translation.hs

```diff
@@ -81,46 +82,54 @@ translateFSM circuitName ctype signs = do
 
 translate :: (Show a, Ord a) => CircuitConcept a -> [a] -> String
 translate circuit signs =
-    case validate signs circuit of
+    case validateInitialState signs circuit of
       Valid -> do
           let allCause = addConsistency (arcs circuit) signs
-              sortedCause = concatMap handleArcs (groupSortOn snd allCause)
+              sortedCause = concatMap handleArcs (NonEmpty.groupAllWith snd (arcLists allCause))
```

Comment:
Add `import Data.List.NonEmpty (NonEmpty, groupAllWith)` to imports and you can skip these qualifiers.